docs/dev/debug_delve.md

```diff
@@ -0,0 +1,89 @@
+
+# assisted-service runs on minikube
+
+## Debug Setup
+
+To debug assisted-service directly from minikube container
+Install minikube setup with assisted-service with one replica
+
+```shell
+export KUBECONFIG=source /root/.kube/config
+kubectl patch deployment assisted-service -n assisted-installer --type json -p="[{\"op\": \"remove\", \"path\": \"/spec/template/spec/containers/0/livenessProbe\"}]"
+oc rollout restart deployment -n assisted-installer assisted-service
+```
+
+Login to minikube node and attach to assisted-service container
+
+```shell
+oc rsh -n assisted-installer [pod/assisted-service-name] 
+# install packages git goland and telnet
+dnf install procps psmisc golang delve iproute iputils telnet net-tools  -y
+```
+
+Before running dlv attach make sure assisted-service sources exists 
+
+```shell
+git clone https://github.com/openshift/assisted-service.git
+# git checkout to the same minikube assisted-service version/tag
+```
+Note! Before attaching to golang binary you should know in case you break/debug 
+in line, the assisted-service pod goes down. in order to start pod again run continue
```

Comment:
You might be able to avoid this by removing the health checks before attempting to debug.

This is probably why the pod goes down.

Reply:
Not sure but if we break for a line and waiting its up and running , once we hit on the lines 
debugger not continue till we force continue .
Are we expecting for active pod when code break in a line  without continue action ?

Reply:
maybe we should build a container with the tooling inside and create a more robust  soultion

Reply:
The doc says "the assisted-service pod goes down"

I'm saying that the cause of this could be that the liveness probes are failing when we break in the debugger.
The point about the changes to the container is also relevant, but I wasn't as much concerned about that in this comment.

Reply:
> The doc says "the assisted-service pod goes down"
> 
> I'm saying that the cause of this could be that the liveness probes are failing when we break in the debugger. The point about the changes to the container is also relevant, but I wasn't as much concerned about that in this comment.

When i hit on break in the debugger i see that pod chnaged
(ocp-edge-qe-venv) [root@ocp-edge127 linchpin-workspace]# oc get pods -n assisted-installer
NAME                                      READY   STATUS    RESTARTS   AGE
assisted-image-service-74954d69df-7ghqs   1/1     Running   0          30h
assisted-installer-ui-5498bcd454-9zzvr    1/1     Running   0          30h
assisted-service-6b8bfd6f44-nn5r6         0/1     Running   0          129m
object-store-558498d8-mv8pn               1/1     Running   0          30h
postgres-5997fdbf9f-w97tk                 1/1     Running   0          30h

Once i continue the debugger it changes to 1/1

Reply:
see the record to be on the safe side
[Screencast from 2024-03-06 19-40-13.webm](https://github.com/openshift/assisted-service/assets/6973510/16b41aaa-b9e3-4e89-97dc-a558ce932915)

Reply:
Right, that's exactly what I'm describing. When you break all the threads stop so we can't respond to readiness checks.

If you remove those health checks you probably won't get this behavior and can debug in peace.

Reply:
Ok , the solution worked and the pod is ready even in a break step - perfect !